### ISSUE TITLE
Accessibility: Missing lists for grouped links

### ISSUE THREAD

**Issue by jason-me** [stakwork/sphinx-tribes-frontend#1049]
## Issue
The list of links in the header, footer, and expanded mobile navigation are missing semantic markup for lists.

## Expected Behavior
- [WCAG 2.2 AA 1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG22/#non-text-content) requires that structure conveyed visually such as lists are also conveyed semantically.
- Lists of links should be grouped in an `<ul>` so that the structure is conveyed semantically
- Screen reader users expect to hear the number of items in a list and their current position (i.e 2 of 3 items) 
## Acceptance Criteria
- [x] Add list semantics to all lists of links in the header, footer, and that are revealed by the disclosure.
- [x] Test with a screen reader JAWS, VoiceOver, NVDA, Orca, Talkback, etc. 
- [x] Ensure the lists are nested within appropriate containers (e.g. `nav` element with an `<ul>`) and that they are announced to screen readers along with the position within the list.

_Linked to tracking issue: #456_

**Comment by jason-me** [comment]
Pending closure via PR: https://github.com/stakwork/sphinx-tribes-frontend/pull/1140

**Comment by jason-me** [comment]
Merged in PR: https://github.com/stakwork/sphinx-tribes-frontend/pull/1140
